### PR TITLE
Reduce memory usage in TF building

### PR DESCRIPTION
**Pull request by Rocketknight1**
This PR reduces the default shape of dummy inputs from (3, 3) to (2, 2). This slightly reduces the memory usage when building TF models, which should hopefully fix some of our pipeline tests.

We could replace the dummy inputs with symbolic tensors, which would mean we could build TF models with 0 memory usage, but this would make TF model building slower (~4X) because it would implicitly compile the model when building, which is probably not an acceptable tradeoff.

cc @ydshieh and @amyeroberts as core maintainer

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
Thanks, the change itself is good for me.

File: src/transformers/modeling_tf_utils.py [hunk 1]
```diff
@@ -1116,16 +1116,16 @@ def dummy_inputs(self) -> Dict[str, tf.Tensor]:
         dummies = {}
         sig = self._prune_signature(self.input_signature)
         for key, spec in sig.items():
-            # 3 is the most correct arbitrary size. I will not be taking questions
-            dummies[key] = tf.ones(shape=[dim if dim is not None else 3 for dim in spec.shape], dtype=spec.dtype)
+            # 2 is the most correct arbitrary size. I will not be taking questions
```
Comment (amyeroberts) [inline src/transformers/modeling_tf_utils.py:1119]: I wish to file this diff as evidence to the contrary #team3

**Review by amyeroberts — APPROVED** [review]
Change LGTM - thanks for updating! 

Happy to merge once @ydshieh gives the 👍  from CI runs

**Comment by Rocketknight1** [comment]
Sorry for the delay - there's an issue with Funnel that wasn't reproducing on my machine. I eventually figured out that the problem is the classic TF one: indices for `tf.gather` are not validated on GPU but are validated on CPU, and so the bug only becomes apparent on CPU. Will fix in just a sec!

**Comment by Rocketknight1** [comment]
Yep, working on it now!

**Comment by ydshieh** [comment]
The `tests/pipelines/test_pipelines_common.py::PipelineUtilsTest::test_load_default_pipelines_tf` run against a list of models, so it's kind normal it fails with other models even some fixes are done previously.

I am OK to trigger the run (a subset) whenever you feel it's time. Otherwise I can show you a modified workflow file for you to trigger manually.

**Comment by ydshieh** [comment]
You can watch it live [here](https://github.com/huggingface/transformers/actions/runs/5191137996/jobs/9358557442). It will take 20-30 min to finish.

**Comment by Rocketknight1** [comment]
Maybe this is a sign that we should transition the dummies to symbolic tensors for those models, even if it's probably too slow for our tests to do it across the whole codebase.

**Pull request merged**